internal/internal_workflow_testsuite.go

```diff
@@ -888,6 +915,21 @@ func (env *testWorkflowEnvironmentImpl) Complete(result *commonpb.Payloads, err
 			}, true /* true to trigger parent workflow to resume to handle child workflow's result */)
 		}
 	}
+
+	// cancel child workflows if their ParentClosePolicy request so.
+	env.maybeCancelChildWorkflows()
+}
+
+func (env *testWorkflowEnvironmentImpl) maybeCancelChildWorkflows() {
+	for _, handle := range env.runningWorkflows {
+		if handle.env.parentEnv != nil &&
+			env.workflowInfo.WorkflowExecution.ID == handle.env.parentEnv.workflowInfo.WorkflowExecution.ID {
+			// current env is parent workflow of handle's workflow
+			if handle.params.ParentClosePolicy == enumspb.PARENT_CLOSE_POLICY_REQUEST_CANCEL {
```

Comment:
What about TERMINATE? I know it doesn't make sense for the root parent workflow, but what about the of the child of the root?

Reply:
Good point. We need to handle that as well. Will do in a follow up PR.